File: WMSWebAPI/Entity/Producto/clsBeProducto_estado.cs
using System.ComponentModel;
using System.ComponentModel.DataAnnotations.Schema;

namespace WMSWebAPI.Entity.Producto
{
    public class clsBeProducto_estado : ICloneable
    {
        [Column("IdEstado")]
        [DisplayName("IdEstado")]
        public int IdEstado { get; set; } = 0;

        [Column("IdPropietario")]
        [DisplayName("IdPropietario")]
        public int IdPropietario { get; set; } = 0;

        [Column("nombre")]
        [DisplayName("nombre")]
        public string Nombre { get; set; } = "";

        [Column("IdUbicacionDefecto")]
        [DisplayName("IdUbicacionDefecto")]
        public int IdUbicacionDefecto { get; set; } = 0;

        [Column("utilizable")]
        [DisplayName("utilizable")]
        public bool Utilizable { get; set; } = false;

        [Column("activo")]
        [DisplayName("activo")]
        public bool Activo { get; set; } = false;

        [Column("user_agr")]
        [DisplayName("user_agr")]
        public string User_agr { get; set; } = "";

        [Column("fec_agr")]
        [DisplayName("fec_agr")]
        public DateTime Fec_agr { get; set; } = DateTime.Now;

        [Column("user_mod")]
        [DisplayName("user_mod")]
        public string User_mod { get; set; } = "";

        [Column("fec_mod")]
        [DisplayName("fec_mod")]
        public DateTime Fec_mod { get; set; } = DateTime.Now;

        [Column("dañado")]
        [DisplayName("dañado")]
        public bool Dañado { get; set; } = false;

        [Column("codigo_bodega_erp")]
        [DisplayName("codigo_bodega_erp")]
        public string Codigo_bodega_erp { get; set; } = "";

        [Column("sistema")]
        [DisplayName("sistema")]
        public bool Sistema { get; set; } = false;

        [Column("dias_vencimiento_clasificacion")]
        [DisplayName("dias_vencimiento_clasificacion")]
        public int Dias_vencimiento_clasificacion { get; set; } = 0;

        [Column("tolerancia_dias_vencimiento")]
        [DisplayName("tolerancia_dias_vencimiento")]
        public int Tolerancia_dias_vencimiento { get; set; } = 0;

        public clsBeProducto_estado() { }

        public object Clone()
        {
            return MemberwiseClone();
        }
    }
}

File: TOMIMSV4/Entity/Mantenimientos/Producto/ProductoEstado/clsBeProducto_estado.vb
<Serializable>
Public Class clsBeProducto_estado
    Implements ICloneable
    Implements IDisposable

    ''' <summary>
    ''' Gets or sets the identifier estado.
    ''' </summary>
    ''' <value>The identifier estado.</value>
    Public Property IdEstado() As Integer = 0
    ''' <summary>
    ''' Gets or sets the identifier propietario.
    ''' </summary>
    ''' <value>The identifier propietario.</value>
    Public Property IdPropietario() As Integer = 0
    ''' <summary>
    ''' Gets or sets the propietario.
    ''' </summary>
    ''' <value>The propietario.</value>
    Public Property Propietario As clsBePropietarios
    ''' <summary>
    ''' Gets or sets the nombre.
    ''' </summary>
    ''' <value>The nombre.</value>
    Public Property Nombre() As String = ""
    ''' <summary>
    ''' Gets or sets the identifier ubicacion defecto.
    ''' </summary>
    ''' <value>The identifier ubicacion defecto.</value>
    Public Property IdUbicacionDefecto() As Integer = 0

    Public Property IdUbicacionBodegaDefecto() As Integer = 0
    ''' <summary>
    ''' Gets or sets a value indicating whether this <see cref="clsBeProducto_estado"/> is utilizable.
    ''' </summary>
    ''' <value><c>true</c> if utilizable; otherwise, <c>false</c>.</value>
    Public Property Utilizable() As Boolean = False
    ''' <summary>
    ''' Gets or sets a value indicating whether this <see cref="clsBeProducto_estado"/> is activo.
    ''' </summary>
    ''' <value><c>true</c> if activo; otherwise, <c>false</c>.</value>
    Public Property Activo() As Boolean = False
    ''' <summary>
    ''' Gets or sets the user_agr.
    ''' </summary>
    ''' <value>The user_agr.</value>
    Public Property User_agr() As String = ""
    ''' <summary>
    ''' Gets or sets the fec_agr.
    ''' </summary>
    ''' <value>The fec_agr.</value>
    Public Property Fec_agr() As Date = Date.Now
    ''' <summary>
    ''' Gets or sets the user_mod.
    ''' </summary>
    ''' <value>The user_mod.</value>
    Public Property User_mod() As String = ""
    ''' <summary>
    ''' Gets or sets the fec_mod.
    ''' </summary>
    ''' <value>The fec_mod.</value>
    Public Property Fec_mod() As Date = Date.Now
    ''' <summary>
    ''' Gets or sets a value indicating whether this <see cref="clsBeProducto_estado"/> is dañado.
    ''' </summary>
    ''' <value><c>true</c> if dañado; otherwise, <c>false</c>.</value>
    Public Property Dañado() As Boolean = False

    Public Property Sistema() As Boolean = False

    ''' <summary>
    ''' #EJC202311150009: Codigo_Bodega_ERP Entity
    ''' </summary>
    ''' <returns></returns>
    Public Property Codigo_Bodega_ERP As String = ""

    ''' <summary>
    ''' #EJC20231118: Dias_Vencimiento_Clasificacion para definir estado del producto en base a la fecha
    ''' </summary>
    ''' <returns></returns>
    Public Property Dias_Vencimiento_Clasificacion As Integer = 0

    ''' <summary>
    ''' #EJC20231118: Tolerancia_DiasVencimiento para definir estado del producto en base a la fecha
    ''' </summary>
    ''' <returns></returns>
    Public Property Tolerancia_Dias_Vencimiento As Integer = 0

    ''' <summary>
    ''' #CKFK20250910: Campo para saber si el estado del producto en la reserva requiere UMBas,
    ''' esto aplica si el cliente tiene definido un estado de producto a importar
    ''' </summary>
    ''' <returns></returns>
    Public Property Reservar_En_UmBas As Boolean = False

    ''' <summary>
    ''' Initializes a new instance of the <see cref="clsBeProducto_estado"/> class.
    ''' </summary>
    Sub New()
    End Sub

    ''' <summary>
    ''' Creates a new object that is a copy of the current instance.
    ''' </summary>
    ''' <returns>A new object that is a copy of this instance.</returns>
    Public Function Clone() As Object Implements System.ICloneable.Clone
        Return MyBase.MemberwiseClone()
    End Function

End Class
